src/ds/address.h

```diff
@@ -119,7 +134,10 @@ namespace snmalloc
 #if __has_builtin(__builtin_align_up)
     return static_cast<T*>(__builtin_align_up(p, alignment));
 #else
-    return pointer_cast<T>(bits::align_up(address_cast(p), alignment));
+    static_assert(
+      aal_supports<IntegerPointers>,
```

Comment:
Since we have this assert in multiple places, it would be good to factor it out and then we can relax the constraint easily later.